### PR TITLE
James 1688 : smtp mtp tests

### PR DESCRIPTION
**Pull request by aduprat**
First draft, missing:
- checking the SMTP process with fakesmtp after the script execution in mpt test
- integration of a validation script in mtp-external-james

### CODE REVIEW

File: mpt/impl/smtp/cassandra/src/test/java/org/apache/james/mpt/smtp/SmtpTestModule.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+                new CassandraMessageModule(),
+                new CassandraDomainListModule(),
+                new CassandraUsersRepositoryModule(),
+                new CassandraRRTModule());
+        cassandraClusterSingleton = CassandraCluster.create(cassandraModuleComposite);
+    }
+
+    @Override
+    protected void configure() {
+        install(Modules.override(Modules.combine(
```
Comment (mbaechler) [inline mpt/impl/smtp/cassandra/src/test/java/org/apache/james/mpt/smtp/SmtpTestModule.java]: would be better to use CassandraJamesServerMain.defaultModule

File: mpt/impl/smtp/cassandra/src/test/java/org/apache/james/mpt/smtp/host/JamesSmtpHostSystem.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    @Override
+    public boolean addUser(String userAtDomain, String password) throws Exception {
+        Preconditions.checkArgument(userAtDomain.contains("@"), "The 'user' should contains the 'domain'");
+        Iterator<String> split = Splitter.on("@").split(userAtDomain).iterator();
+        String user = split.next();
+        String domain = split.next();
+
+        domainList.addDomain(domain);
+        usersRepository.addUser(userAtDomain, password);
+        recipientRewriteTable.addAddressMapping(user, domain, "ray@yopmail.com");
```
Comment (mbaechler) [inline mpt/impl/smtp/cassandra/src/test/java/org/apache/james/mpt/smtp/host/JamesSmtpHostSystem.java]: probably nothing to do here !

File: mpt/impl/smtp/core/src/main/java/org/apache/james/mpt/smtp/ForwardSmtpTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    @Before
+    public void setUp() throws Exception {
+        InetAddress containerIp = InetAddresses.forString(fakeSmtp.getContainerIp());
+        dnsService.registerRecord("yopmail.com", new InetAddress[]{containerIp}, ImmutableList.of("yopmail.com"), ImmutableList.of());
+        super.setUp();
+    }
+
+    @Test
+    public void authenticateShouldWork() throws Exception {
+        WatchService watcher = FileSystems.getDefault().newWatchService();
```
Comment (mbaechler) [inline mpt/impl/smtp/core/src/main/java/org/apache/james/mpt/smtp/ForwardSmtpTest.java]: we were supposed to use another fake smtp that talk rest for assertions

File: mpt/impl/smtp/cassandra/src/test/java/org/apache/james/mpt/smtp/SmtpTestModule.java (hunk 1 above)
Comment (aduprat) [inline mpt/impl/smtp/cassandra/src/test/java/org/apache/james/mpt/smtp/SmtpTestModule.java]: OK

File: mpt/impl/smtp/cassandra/src/test/java/org/apache/james/mpt/smtp/host/JamesSmtpHostSystem.java (hunk 2 above)
Comment (aduprat) [inline mpt/impl/smtp/cassandra/src/test/java/org/apache/james/mpt/smtp/host/JamesSmtpHostSystem.java]: Right, moved in setup

File: mpt/impl/smtp/core/src/main/java/org/apache/james/mpt/smtp/ForwardSmtpTest.java (hunk 3 above)
Comment (aduprat) [inline mpt/impl/smtp/core/src/main/java/org/apache/james/mpt/smtp/ForwardSmtpTest.java]: OK

**Comment by mbaechler** [comment]
merged from another PR

**Pull request closed without merging**